Add variables for link colors

jenny-heath opened the pull request:
We want to be able to overwrite link colors more easily for our project so we added variables for default, hover state, active state, etc.

Co-authored-by: Molly T-M <mollyt@codeforamerica.org>

jenny-heath commented on Gemfile.lock:
```diff
@@ -285,5 +285,8 @@ DEPENDENCIES
   selenium-webdriver
   uglifier
 
+RUBY VERSION
```
not sure what's up with this? we can remove it but wanted to see if anyone knows why this was added

hartsick replied:
it [looks like](https://devcenter.heroku.com/articles/ruby-versions#ruby-version-specifiers) it comes with declaring the ruby version in the Gemfile, which we do. I believe this would also only impact the I don't think it's harmful to have it in (especially because this would impact dev and the app, not host applications), but that's only through doing some light research.

lkogler replied:
This is probably my fault. A while ago the heroku deployment broke because it was trying to use the wrong ruby version, so I [added the ruby version to the Gemfile](https://github.com/codeforamerica/honeycrisp-gem/commit/c4bfbd55192cc937065dd991426035e41331e678) to fix that issue, but it looks like I forgot to commit the corresponding change to the Gemfile.lock. Oops!

hartsick commented:
@jenny-heath I just pushed an addition to the styleguide documentation that is a link styled as a button, which I think is the case we need to consider when making these changes. You can view it by running `rails s`, visiting http://localhost:3000/cfa/styleguide#atoms-buttons. it's the "Links as buttons" section.

Right now it looks like this PR breaks those buttons (can compare with cherry-picking the documentation commit master). Also, I think it might be worth looking at the @outline mixin that's included in the a tag CSS to make sure we're covering all the link colors with our new variables.

feel free to ping me on Slack if you have any questions or want to pair on this

hartsick commented on app/views/examples/atoms/_buttons.html.erb:
```diff
@@ -1,49 +1,61 @@
+</div>
 
-<p class="pattern__explanation">
+<div class="h4">Links as buttons</div>
```
this is the code that is currently not looking like I'd expect

jenny-heath commented:
Update on this: We figured out how to fix the styling of links-as-buttons. Turns out they looked weird because in the button classes we don't have rules for visited/active even though we added visited/active to the link styling (so it was taking the color from `a:visited`). We added these rules to the buttons, and in fact they are a duplicate of existing rules that specify hover color on buttons in order not to collide with `a:hover`.
This also means that we have a way to not clobber our own button styling (just add these same rules in our application), and no longer need to change the styleguide in order to customize our link colors in a relatively easy way. However, we made this PR in part because it could be a useful addition to the styleguide. There are two drawbacks that I can see to including this in honeycrisp:
1. `variables` needs to be imported before `cfa_styleguide_main` in our application for this custom link color variable thing to work. this doesn't seem obvious to me and i'm not sure where to document it for folks who want to use this
2. maybe we don't want to enable changing these styles, or we don't want to enable it in this way

Happy to either merge this or drop it, depending on whether it seems like something others can/will use

hartsick commented:
@jenny-heath Thanks for digging into!

For this PR, I'd be pro including the splitting out those variables because they seem useful, but we'd have to make changes to this code to fix the links-as-buttons styling before so we don't break existing projects before I'd be comfortable merging. Seems like that's some additional work, so look to you whether it's something you'd care to do!

For the variables, agree overriding isn't intuitive. We did include it in the documentation here https://github.com/codeforamerica/honeycrisp-gem#usage (see step 5), but wondering if you have any ideas as to how we could make that more visible, either in the documentation or in the hosted app?

hartsick commented:
@jenny-heath oh oops, I'm sorry! I just saw there's another commit on this to address the links. thanks for doing that. I'll take a look at that when I have time this week and will let you know if I have any questions!

lkogler commented:
> @jenny-heath Thanks for digging into!
> 
> For this PR, I'd be pro including the splitting out those variables because they seem useful, but we'd have to make changes to this code to fix the links-as-buttons styling before so we don't break existing projects before I'd be comfortable merging. Seems like that's some additional work, so look to you whether it's something you'd care to do!
> 
> For the variables, agree overriding isn't intuitive. We did include it in the documentation here https://github.com/codeforamerica/honeycrisp-gem#usage (see step 5), but wondering if you have any ideas as to how we could make that more visible, either in the documentation or in the hosted app?

Ahh, I have had trouble figuring out how to do this in the past! I do think it would be helpful to highlight it more in the documentation. Maybe we could move steps 5 and 6 to a new section, named "Customizing styles" or something like that. I think the fact that they're buried in a long list and marked "optional" makes them easy to skim over and miss.

lkogler commented:
> @jenny-heath I just pushed an addition to the styleguide documentation that is a link styled as a button, which I think is the case we need to consider when making these changes. You can view it by running rails s, visiting http://localhost:3000/cfa/styleguide#atoms-buttons. it's the "Links as buttons" section.
> 
> Right now it looks like this PR breaks those buttons (can compare with cherry-picking the documentation commit master). Also, I think it might be worth looking at the @outline mixin that's included in the a tag CSS to make sure we're covering all the link colors with our new variables.
> 
> feel free to ping me on Slack if you have any questions or want to pair on this

This commit also addresses open issue: https://github.com/codeforamerica/honeycrisp-gem/issues/165

lkogler commented:
Tests are passing locally for me, but not on CI. I wonder if this could be fixed by disabling concurrency as described here: https://github.com/sass/sassc-rails/issues/121

lkogler commented:
Hmm, seems to be an intermittent CI failure -- It passed when I re-ran it. Let's keep an eye on it.

lkogler commented:
Just added a couple more commits - one to make the documentation change I suggested and another for some minor cleanup of the button documentation section.
